OpenTokSamples/7.Multiparty-ConstraintLayout/src/main/java/com/opentok/android/samples/multiparty_constraintlayout/MainActivity.java

```diff
@@ -0,0 +1,328 @@
+
+    private ConstraintLayout mContainer;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        Log.d(TAG, "onCreate");
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mContainer = (ConstraintLayout) findViewById(R.id.main_container);
+
+        requestPermissions();
+    }
+
+    @Override
+    protected void onResume() {
+        Log.d(TAG, "onResume");
+
+        super.onResume();
+
+        if (mSession == null) {
+            return;
+        }
+        mSession.onResume();
+    }
+
+    @Override
+    protected void onPause() {
+        Log.d(TAG, "onPause");
+
+        super.onPause();
+
+        if (mSession == null) {
+            return;
+        }
+        mSession.onPause();
+
+        if (isFinishing()) {
+            disconnectSession();
+        }
+    }
+
+    @Override
+    protected void onDestroy() {
+        Log.d(TAG, "onDestroy");
+
+        disconnectSession();
+
+        super.onDestroy();
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+
+        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this);
+    }
+
+    @Override
+    public void onPermissionsGranted(int requestCode, List<String> perms) {
+        Log.d(TAG, "onPermissionsGranted:" + requestCode + ":" + perms.size());
+    }
+
+    @Override
+    public void onPermissionsDenied(int requestCode, List<String> perms) {
+        Log.d(TAG, "onPermissionsDenied:" + requestCode + ":" + perms.size());
+
+        if (EasyPermissions.somePermissionPermanentlyDenied(this, perms)) {
+            new AppSettingsDialog.Builder(this, getString(R.string.rationale_ask_again))
+                    .setTitle(getString(R.string.title_settings_dialog))
+                    .setPositiveButton(getString(R.string.setting))
+                    .setNegativeButton(getString(R.string.cancel), null)
+                    .setRequestCode(RC_SETTINGS_SCREEN_PERM)
+                    .build()
+                    .show();
+        }
+    }
+
+    private void startPublisherPreview() {
+        mPublisher = new Publisher.Builder(this).name("publisher").build();
+
+        mPublisher.setPublisherListener(this);
+        mPublisher.setStyle(BaseVideoRenderer.STYLE_VIDEO_SCALE, BaseVideoRenderer.STYLE_VIDEO_FILL);
+        mPublisher.startPreview();
+    }
+
+    @AfterPermissionGranted(RC_VIDEO_APP_PERM)
+    private void requestPermissions() {
+        String[] perms = {
+                Manifest.permission.INTERNET,
+                Manifest.permission.CAMERA,
+                Manifest.permission.RECORD_AUDIO
+        };
+        if (EasyPermissions.hasPermissions(this, perms)) {
+            startPublisherPreview();
```

Comment:
nit: If you are not calling/using the `startPublisherPreview()` method any other place there is no need for having a method. IMHO The content of `startPublisherPreview()` is small enough to leave so readability is ok is you do not have the method. Anyway If you do not want to get rid of `startPublisherPreview()` is okay too.